csharp/src/Ice/Internal/IPEndpointI.cs

```diff
@@ -150,21 +119,11 @@ public override List<EndpointI> expandHost(out EndpointI publishedEndpoint)
             instance_.preferIPv6(),
             true);
 
-        if (addresses.Count == 1)
```

Comment:
This logic is not good, as we often end-up returning a DNS name (such as localhost) for the IncomingConnectionFactory endpoint. We do want actual IP addresses for these endpoints.